stats/stats.go

```diff
@@ -393,21 +401,35 @@ func (h *Histogram) Clone() *Histogram {
 // CopyFrom sets the content of this object to a copy of the src.
 func (h *Histogram) CopyFrom(src *Histogram) {
 	h.Counter = src.Counter
-	// we don't copy offset/divider as this assumes compatible src/dest
-	for i := 0; i < len(h.Hdata); i++ {
-		h.Hdata[i] += src.Hdata[i]
+	h.CopyHDataFrom(src)
+}
+
+// CopyHDataFrom appends histogram data values to this object from the src.
+// Src histogram data values will be appended according to this object's
+// offset and divider
+func (h *Histogram) CopyHDataFrom(src *Histogram) {
+	if h.Divider == src.Divider && h.Offset == src.Offset {
+		for i := 0; i < len(h.Hdata); i++ {
+			h.Hdata[i] += src.Hdata[i]
+		}
+		return
+	}
+
+	hData := src.Export([]float64{})
+	for _, v := range hData.Data {
+		if v.Count == 1 {
+			h.appendRecordToHistogram(v.Start)
```

Comment:
that's not correct, if the value is neither min nor max, there could be a single count in the bucket and the best value is still the midpoint, not the first one